File: Python/BOJ/연결리스트/1406.py
import sys

left = list(sys.stdin.readline().rstrip())
right = []
n = len(left)
m = int(sys.stdin.readline().rstrip())

for _ in range(m):
  inputList = list(sys.stdin.readline().rstrip().split())
  command = inputList[0]
  if command == "L":
    if left:
      right.append(left.pop())
  elif command == "D":
    if right:
      left.append(right.pop())
  elif command == "B":
    if left:
      left.pop()
  elif command == "P":
    left.append(inputList[1])

left += reversed(right)
print("".join(left))

File: Swift/BOJ/문제집/연결리스트/1406.swift
//
//  1406.swift
//  Algorithm
//
//  Created by 유정주 on 2023/06/27.
//

import Foundation

var left = readLine()!
let m = Int(readLine()!)!
var right = ""

for _ in 0..<m {
    var command = readLine()!
    switch command {
    case "L":
        if !left.isEmpty {
            right.append(left.removeLast())
        }
    case "D":
        if !right.isEmpty {
            left.append(right.removeLast())
        }
    case "B":
        if !left.isEmpty {
            left.removeLast()
        }
    default:
        let value = String(command.last!)
        left.append(value)
    }
}

print(left + right.reversed())
